tools/c7n_azure/c7n_azure/resources/access_control.py

```diff
@@ -96,3 +141,88 @@ class UserRole(RelatedResourceFilter):
 
     RelatedResource = "c7n_azure.resources.access_control.RoleDefinition"
     RelatedIdsExpression = "properties.roleDefinitionId"
+
+
+@RoleAssignment.filter_registry.register('resource-access')
+class ResourceAccessFilter(Filter):
+    """Filters role assignments that have access to a certain
+    type of azure resource.
+
+    :Example:
+
+        .. code-block:: yaml
+
+            policies:
+               - name: assignments-by-azure-resource
+                 resource: azure.roleassignment
+                 filters:
+                    - type: resource-access
+                      relatedResource: azure.vm
+    """
+
+    schema = type_schema(
+        'resource-access',
+        relatedResource={'type': 'string'},
+        required=['relatedResource']
+    )
+
+    def __init__(self, data, manager=None):
+        super(ResourceAccessFilter, self).__init__(data, manager)
+        resource_type = self.data['relatedResource']
+        self.factory = Azure.resources.get(
+            resource_type.rsplit('.', 1)[-1])
+
+    def get_related(self):
+        ctx = ExecutionContext(local_session(Session), self.data, Config.empty())
+
+        manager = self.factory(ctx, self.data)
+        return manager.source.get_resources(None)
+
+    def process(self, resources, event=None):
+        related_resources = self.get_related()
+        s = local_session(self.manager.session_factory)
+        assignment_ids = []
+        headers = {
+            'Content-type': 'application/json',
+            'Authorization': 'Bearer %s' % (s.get_bearer_token())
+        }
+        for resource in related_resources:
+            url = (
+                'https://management.azure.com%s/'
+                'providers/Microsoft.Authorization/roleAssignments?api-version=2015-07-01'
+                % (resource['id'])
+            )
```

Comment:
Impressive but not sure if OK :)  If we wanted to do this we'd need to lookup the correct API version programmatically and probably make a utility to make these calls.  

Anyway it looks like the `list_for_scope` builds the exact same URL if you don't like the parameter structure of `list_for_resource`

It does this:
`/{scope}/providers/Microsoft.Authorization/roleAssignments`

https://github.com/Azure/azure-sdk-for-python/blob/5444f712516faf65dcb5e38a45bcbbbac89a9790/azure-mgmt-authorization/azure/mgmt/authorization/v2015_07_01/operations/role_assignments_operations.py#L674

Reply:
No I previously had the client that called list_for_resource here and it worked fine except for 2 things: parent path (which is nothing most of the time), and the test for py27 vs py36 would fail because the client makes different calls (one with an extra slash and the other not). I know the code to get the parent path, it's just quite cumbersome but it's possible. I can swap the code back to using the client, add the code to get the parent path, and just remove the test since it's impossible to make it pass for both python versions.

Reply:
Those sound like issues with `list_for_resource` but they seem to provide `list_for_scope` to do exactly what you want.

Reply:
Oh perfect, will swap out!